tests/unittests/PartitionerTest.cpp

```diff
@@ -526,40 +526,91 @@ static void createSimpleModule(Module &mod) {
-    DAGListTy myList = std::move(partitioner.getPartitionResult());
-    ASSERT_EQ(mod_.getFunctions().size(), 3);
-    ASSERT_EQ(myList.size(), 1);
-    ASSERT_TRUE(checkSaveNode(mod_));
-
-    for (auto &dag : myList) {
-      for (auto &node : dag.nodes) {
-        // Although the saturateHost is set true, no saturating the host in
-        // heterogeneous partiton.
-        ASSERT_EQ(node->logicalDevices.size(), 1);
-      }
+  createSimpleModule(mod_);
+  BackendWithoutSub backendWithoutSub1, backendWithoutSub2;
+  BackendWithoutMul backendWithoutMul1, backendWithoutMul2;
+  // Create two backends which support different ops, then do the partition by
+  // assigning the ops to the corresponding abackends.
+  std::vector<Backend *> backends;
+  backends.emplace_back(&backendWithoutMul1);
+  backends.emplace_back(&backendWithoutMul2);
+  backends.emplace_back(&backendWithoutSub1);
+  backends.emplace_back(&backendWithoutSub2);
+  std::vector<DeviceInfo> devices = {{3072, "Interpreter"},
+                                     {3072, "Interpreter"},
+                                     {3072, "CPU"},
+                                     {3072, "CPU"}};
+  auto partitioner =
+      Partitioner(&mod_, devices, backends, /* saturateHost */ true);
+  CompilationContext cctx;
+  auto err = partitioner.Partition(cctx);
+  EXPECT_FALSE(errToBool(std::move(err)));
+  DAGListTy myList = std::move(partitioner.getPartitionResult());
+  ASSERT_EQ(mod_.getFunctions().size(), 3);
+  ASSERT_EQ(myList.size(), 1);
+  ASSERT_TRUE(checkSaveNode(mod_));
+
+  for (auto &dag : myList) {
+    for (auto &node : dag.nodes) {
+      // Although the saturateHost is set true, no saturating the host in
+      // heterogeneous partiton.
+      ASSERT_EQ(node->logicalDevices.size(), 1);
+    }
+  }
+  mod_.clear();
+}
+
+/// Test pre-defined non-supported ops used for choosing backend in
+/// Heterogeneous Partition.
+TEST_F(PartitionerTest, ConfigHeterogeneousPartitioning1) {
+  createSimpleModule(mod_);
+  std::vector<DeviceInfo> devices = {{3072, "Interpreter", "Mul"},
+                                     {3072, "Interpreter", "Mul"},
+                                     {3072, "CPU", "Sub"},
+                                     {3072, "CPU", "Sub"}};
+  auto partitioner = Partitioner(&mod_, devices);
+  CompilationContext cctx;
+  auto err = partitioner.Partition(cctx);
+  EXPECT_FALSE(errToBool(std::move(err)));
+  DAGListTy myList = std::move(partitioner.getPartitionResult());
+  ASSERT_EQ(mod_.getFunctions().size(), 3);
+  ASSERT_EQ(myList.size(), 1);
+  ASSERT_TRUE(checkSaveNode(mod_));
+  for (auto &dag : myList) {
+    for (auto &node : dag.nodes) {
+      // Although the saturateHost is set true, no saturating the host in
+      // heterogeneous partiton.
+      ASSERT_EQ(node->logicalDevices.size(), 1);
     }
-    mod_.clear();
   }
+  mod_.clear();
+}
+
+/// Test pre-defined supported ops used for choosing backend in
+/// Heterogeneous Partition.
```

Comment:
It is not obvious to me from the code and comments below how the supported ops are tested.